src/messaging/ExchangeContext.cpp

```diff
@@ -129,11 +138,13 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
     const Transport::PeerAddress * peerAddress = GetSessionHandle().GetPeerAddress(mExchangeMgr->GetSessionManager());
     // Treat unknown peer address as "not UDP", because we have no idea whether
     // it's safe to do MRP there.
-    bool isUDPTransport                = peerAddress && peerAddress->GetTransportType() == Transport::Type::kUdp;
+    bool isUDPTransport = peerAddress && peerAddress->GetTransportType() == Transport::Type::kUdp;
+
+    // this check is ignored by the ExchangeMsgDispatch if !AutoRequestAck()
     bool reliableTransmissionRequested = isUDPTransport && !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck);
 
     // If a response message is expected...
-    if (sendFlags.Has(SendMessageFlags::kExpectResponse))
+    if (sendFlags.Has(SendMessageFlags::kExpectResponse) && !IsGroupExchangeContext())
```

Comment:
Instead of doing this, can we actually clear the `kExpectResponse` flag fairly early in this function if it is a group exchange instead? This ensures that other (if any) bits of logic that act on that flag will all behave correctly as well, as opposed to patching up those other places in the future as well to deal with group exchange specifically.

Reply:
`sendFlags` is only use within this function. Clearing a local parameter that is used twice made less sense than to actually check where it is used. Moreover as `sendFlags` is a `const` parameter I don't think any changes can be made directly to it without changing the function definition, Which is not worth it in my opinion to save a single check.

Reply:
Fair enough